.github/workflows/ci.yml

```diff
@@ -0,0 +1,102 @@
+name: Run CI
+on:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md' # Do not need to run CI for markdown changes.
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - '**.md'
+
+jobs:
+  build-linux:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        resolver: ["lts-16.31", "lts-18.28", "lts-19.33", "lts-20.11"]
+
+    services:
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # If you only need the current version keep this.
+
+      - uses: ./.github/actions/setup-cache
+      - uses: ./.github/actions/ci
+        with:
+          resolver: ${{ matrix.resolver }}
+
+  build-macosx:
+    runs-on: macos-latest
+
+    strategy:
+      matrix:
+        resolver: ["lts-16.31", "lts-18.28", "lts-19.33", "lts-20.11"]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # If you only need the current version keep this.
+
+      - name: Install os dependencies
+        run: brew install pcre
+
+      - uses: ./.github/actions/setup-cache
+      - uses: ./.github/actions/ci
+        with:
+          resolver: ${{ matrix.resolver }}
+          run_tests: false # We don't have redis running here so the tests will fail
+
+  quality-checks:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # If you only need the current version keep this.
+
+      - uses: ./.github/actions/setup-cache
+        with:
+          ghc-version: 9.4.7
```

Comment:
One of these ghc-versions quotes, the other does not. Functionally I think they both end up strings, but being different gives me pause.